cmd/juju/status/history.go

```diff
@@ -67,9 +62,30 @@ func (c *statusHistoryCommand) Info() *cmd.Info {
 	}
 }
 
+func supportedHistoryKindTypes() string {
+	supported := set.NewStrings()
+	for k, _ := range status.AllHistoryKind() {
+		supported.Add(string(k))
+	}
+	return strings.Join(supported.SortedValues(), "|")
+}
+
+func supportedHistoryKindDescs() string {
+	types := status.AllHistoryKind()
+	supported := set.NewStrings()
+	for k, _ := range types {
+		supported.Add(string(k))
+	}
```

Comment:
do we need a deduplication set vs just using a []string and then sort.Strings()

Do we want them strictly alphabetical instead of being grouped contextually? (eg juju-unit next to unit)

Maybe it's fine

Reply:
It's not really a deduplication since I'd need to iterate over keys specifically in order to do sorting. Does it really matter if it's []string or set.Strings? I did not think so.

As for grouping contextually, I think it works really well with a bigger/longer set... Apart from that, most of people operate best on alphabetically order. The preference would also depend on how much knowledge of the subject individual user has - are they just exploring and reading help or are they looking for something specific...

I'll land as is and we can iterate based on feedback \o/